src/hits/__tests__/hits.spec.ts

```diff
@@ -4,38 +4,87 @@ import { NgAisHighlight } from '../../highlight/highlight';
 const defaultState = {
   hits: [
-    { name: 'foo', description: 'foo' },
-    { name: 'bar', description: 'bar' },
-    { name: 'foobar', description: 'foobar' },
-    { name: 'barfoo', description: 'barfoo' },
+    { objectID: '1', name: 'foo', description: 'foo' },
+    { objectID: '2', name: 'bar', description: 'bar' },
+    { objectID: '3', name: 'foobar', description: 'foobar' },
+    { objectID: '4', name: 'barfoo', description: 'barfoo' },
   ],
-  results: {},
+  results: { nbPages: 50, page: 0, hitsPerPage: 20 },
 };
 
-const render = createRenderer({
-  defaultState,
-  template: '<ais-hits></ais-hits>',
-  TestedWidget: NgAisHits,
-  additionalDeclarations: [NgAisHighlight],
-});
-
 describe('Hits', () => {
   it('renders markup without state', () => {
+    const render = createRenderer({
+      defaultState,
+      template: '<ais-hits></ais-hits>',
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
     const fixture = render();
     expect(fixture).toMatchSnapshot();
   });
 
   it('renders markup with state', () => {
+    const render = createRenderer({
+      defaultState,
+      template: '<ais-hits></ais-hits>',
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
     const fixture = render({});
     expect(fixture).toMatchSnapshot();
   });
 
   it('should apply `transformItems` if specified', () => {
+    const render = createRenderer({
+      defaultState,
+      template: '<ais-hits></ais-hits>',
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
     const fixture = render({});
     fixture.componentInstance.testedWidget.transformItems = items =>
       items.map(item => ({ ...item, name: `transformed - ${item.name}` }));
     fixture.componentInstance.testedWidget.updateState(defaultState, false);
     fixture.detectChanges();
     expect(fixture).toMatchSnapshot();
   });
+
+  it('should expose hits to the passed template', () => {
+    const render = createRenderer({
+      defaultState,
+      template: `
+          <ais-hits>
+            <ng-template let-hits="hits">
+              <ul>
+                <li *ngFor="let hit of hits">
+                  {{hit.name}}
+                </li>
+              </ul>
+            </ng-template>
+          </ais-hits>
+        `,
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = render({});
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should expose results to the passed template', () => {
+    const render = createRenderer({
+      defaultState,
+      template: `
+          <ais-hits>
+            <ng-template let-results="results">
+              {{ results | json }}
+            </ng-template>
+          </ais-hits>
+        `,
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = render({});
+    expect(fixture).toMatchSnapshot();
```

Comment:
There's no way to test that the fixture's template was called with `results` instead of using a snapshot?

Reply:
I found a way but it's far too complicated.

```ts
    // 1. create mock component and extract instance in scope.
    let mockInstance;
    @Component({
      selector: 'mock-component',
      template: ``,
    })
    class MockComponent implements OnInit {
      @Input() hits: any;
      @Input() results: any;
      @Input() insights: any;
      ngOnInit() {
        mockInstance = this; 
      }
    }

    // 2. pass hits, results and other connector properties to template
    const render = createRenderer({
      defaultState,
      template: `
          <ais-hits>
            <ng-template let-hits="hits" let-results="results" let-insights="insights">
              <mock-component [hits]="hits" [results]="results" [insights]="insights"></mock-component>
            </ng-template>
          </ais-hits>
        `,
      TestedWidget: NgAisHits,
      additionalDeclarations: [NgAisHighlight, MockComponent],
    });

    const state = {
      ...defaultState,
      insights: jest.fn(),
    };

    // 3. trigger render
    render(state);

    // 4. assert
    expect(mockInstance.hits).toEqual(state.hits);
    expect(mockInstance.results).toEqual(state.results);
    expect(mockInstance.insights).toEqual(state.insights);
```

Another technique involved digging into what seems to be private properties of the fixture. 

For small json objects, snapshots are more straight forward.

Reply:
That'd be nice if we could reuse this in a generic way!

Reply:
I'll keep looking into making this more reusable.
I guess we'll have to as we progress in allowing `ng-template` on more widgets and making it the first way to customize their UI.